sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/ServiceTag.java

```diff
@@ -0,0 +1,136 @@
+     */
+    public static final ServiceTag VIRTUAL_NETWORK = new ServiceTag("VirtualNetwork", true, false, false, false);
+
+    /**
+     * Get service tag from name.
+     *
+     * @param serviceTagName the name of the service tag, can be with or without region
+     * @return service tag, null if invalid
+     */
+    public static ServiceTag fromName(String serviceTagName) {
+        if (serviceTagName == null) {
+            return null;
+        }
+        String[] tagParts = serviceTagName.split(Pattern.quote("."));
+        if (tagParts.length == 1) { // root Tag
+            if (serviceTagName.contains(".")) {
+                return null;
+            }
+            return VALUES_BY_NAME.get(tagParts[0]);
+        } else if (tagParts.length == 2) { // Tag.Region
+            ServiceTag rootTag = VALUES_BY_NAME.get(tagParts[0]);
+            if (rootTag == null || CoreUtils.isNullOrEmpty(tagParts[1])) {
+                return null;
+            }
+            return rootTag.withRegion(Region.fromName(tagParts[1]));
+        } else { // unknown
+            return null;
+        }
+    }
+
+    /**
+     * Get regional service tag.
+     *
+     * @param region the region of the regional service tag, will be <strong>omitted</strong> if
+     *               <p>It is null.</p>
+     *               <p>The root service tag either cannot be regional or is already regional.</p>
+     * @return regional service tag
+     */
+    public ServiceTag withRegion(Region region) {
+        if (region == null || !this.regional) {
+            return this;
+        }
+        String tagName = getServiceTagNameWithRegion(this.name, region);
+        ServiceTag tag = VALUES_BY_NAME.get(tagName);
+        return tag != null ? tag : new ServiceTag(getServiceTagNameWithRegion(tagName, region),
+            this.inbound, this.outbound, true, this.firewall);
+    }
+
+    /** @return name of the service tag */
+    public String name() {
+        return name;
+    }
+
+    // service tag name
+    private final String name;
+    // can use in inbound rule
+    private final boolean inbound;
+    // can use in outbound rule
+    private final boolean outbound;
+    // can be regional, e.g. Storage.WestUS
+    private final boolean regional;
+    // can use with Azure Firewall
+
+    private final boolean firewall;
+
+    private ServiceTag(String name, boolean inbound, boolean outbound, boolean regional, boolean firewall) {
+        this.name = name;
+        this.inbound = inbound;
+        this.outbound = outbound;
+        this.regional = regional;
+        this.firewall = firewall;
+        VALUES_BY_NAME.put(name, this);
+    }
+
+    private static String getServiceTagNameWithRegion(String serviceTagName, Region region) {
+        if (alreadyRegional(serviceTagName)) {
+            return serviceTagName;
+        }
+        return String.format("%s.%s", serviceTagName, region.label().replaceAll(Pattern.quote(" "), ""));
```

Comment:
you really sure it is label without space, not the name?

Reply:
I'm pretty certain this is the case. I checked several regional service tags, and the names are like `Storage.WestUS`(label without space), not `Storage.westus` (region name).

A list of service tags (including regional) can be downloaded [here](https://www.microsoft.com/en-us/download/details.aspx?id=56519).

Reply:
OK, we might need some advice from service.

My point
1. region name is case insensitive (but I am not sure about the serviceTag)
2. manipulation on label then send is pretty strange (sending label is already a warning sign, that it requires manipulating is another)
3. if region is unknown, it can be created via "Region.fromName" from service response, there is no guarantee that it takes a label (with uppercase and space). It is more likely created from a name (without casing and space).

Reply:
Got it. 

I tried to create with `Storage.westus`,  it succeeded. I also tried with `Storage.westus5678` (which is an invalid tag) and got 400 failure.

So using the region name should be fine.

Reply:
If the whole serviceTag is case-insensitive, you might need to handle it with this (e.g. when looking up the map etc.)

Reply:
The whole serviceTag is case-insensitive!

Tried with `actiongroup` and `storage.westus` and they both worked.

Amazing. Will update the treatment, thanks!

Reply:
Maybe still worth try reaching one service guy for double confirmation.

Reply:
Sure. I'll check with them.